### PR TITLE
GreeterCard Component

### PR DESCRIPTION
**Pull request by Charca**
**[Trello Card](https://trello.com/c/V0KspGIW/78-3-hsds-notificationcard-content)**

This PR adds the `GreeterCard` component which provides the Greeter UI to be used in embed.

## Demo

Demo

### CODE REVIEW

File: .storybook/storybook.css [hunk 1]
```diff
@@ -1,3 +1,5 @@
+@import url('https://fonts.googleapis.com/css?family=Barlow');
```
Comment (Charca) [inline .storybook/storybook.css]: I've added this so I could preview the Greeter using the right font in Storybook. The font is obviously not included in the hsds-react package.

**Comment by ItsJonQ** [comment]
cc'ing @digitaldesigner for visual/interaction QA 🤗

**Comment by Charca** [comment]
@ItsJonQ any ideas why Travis build is failing? All tests seem to be passing, but for some reason it's exiting with an error :(

**Comment by ItsJonQ** [comment]
@Charca Test coverage droppped:

```
 components/GreeterCard            |    74.58 |      100 |    77.78 |       75 |                   |
  GreeterCard.tsx                  |      100 |      100 |      100 |      100 |                   |
  GreeterCard.utils.ts             |      100 |      100 |      100 |      100 |                   |
  GreeterCardButton.tsx            |        0 |      100 |        0 |        0 |... ,9,10,11,12,18 |
  index.ts                         |        0 |      100 |      100 |        0 |             1,3,4 |
```

😅

File: .storybook/storybook.css (hunk 1 above)
Comment (ItsJonQ) [reply .storybook/storybook.css]: Perfect!

File: src/components/GreeterCard/GreeterCard.css.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import Animate from '../Animate'
+import Button from '../Button'
+import Heading from '../Heading'
+import Text from '../Text'
+import baseStyles from '../../styles/resets/baseStyles.css'
+import { FONT_FAMILY } from '../../styles/configs/constants'
+import { getColor } from '../../styles/utilities/color'
+
+const fontFamily = `
+  font-family: 'Barlow', ${FONT_FAMILY} !important;
```
Comment (ItsJonQ) [inline src/components/GreeterCard/GreeterCard.css.ts]: Awesome!!! Font family, extended

**Comment by digitaldesigner** [comment]
Awesome work @ItsJonQ! Some minor points:

- Top right border radius should be same as top left (that's entirely my fault as my Library component was fucked up, I'm sorry)
- Additional 5px padding bottom when there's no action (bottom)
- Just a question about the focus state of the button (with the thin blue border, is this what we have elsewhere in Beacon? I can't remember)

**Comment by ItsJonQ** [comment]
🍐 'ed with @Charca on the review! Overall, absolutely fantastic! 

I'm making minor adjustments to certain styles and component structure things. Also ensure the button has proper theming support, out of the box!

screen recording 2019-02-27 at 04 53 pm

Got the story working with some theme colours

**Comment by ItsJonQ** [comment]
@digitaldesigner 

> Just a question about the focus state of the button (with the thin blue border, is this what we have elsewhere in Beacon? I can't remember)

Ah! The thinner border is coming from the older `Button` (v1). I've replacing it with the much better, theme-aware `Button` (v2). This is the one with the fancy 2px blue focus ring 👍

**Comment by ItsJonQ** [comment]
@digitaldesigner + @Charca Pushed up my refactor changes!!!

I'll need to write some more tests. Travis will break, haha. @digitaldesigner , I'll also make the UI adjustments you mentioned here:

https://github.com/helpscout/hsds-react/pull/506#issuecomment-468028006

**Comment by ItsJonQ** [comment]
@digitaldesigner + @Charca 

Alrighty! Made the UI adjustments. Borders are now a uniform `8px` all around.
Additional `5px` added to the bottom, if no action. See GIF below.

With action, offset is `21px` (1px for the border).
Without, offset is `26px`.

screen recording 2019-02-27 at 05 51 pm

Also set up animation properties for the component and Storybook for finer grain QA/testing.

screen shot 2019-02-27 at 5 52 39 pm

**Comment by ItsJonQ** [comment]
## Demo link

https://deploy-preview-506--hsds-react.netlify.com/?selectedKind=GreeterCard&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

**Comment by digitaldesigner** [comment]
@ItsJonQ Should that button have a bigger font-weight?

**Comment by Charca** [comment]
@ItsJonQ I believe bottom-right corner (or bottom-left on left aligned Beacon) should still have the smaller border-radius.

**Comment by ItsJonQ** [comment]
@digitaldesigner I can bump it! The button is using the same styles as the normal ones, which is `500`. I guess `500` looks a little different for the Barlow font. I'll bump it 👍 

@Charca Oh yes! You're right! Thanks for the heads up. Will fix

**Comment by ItsJonQ** [comment]
screen shot 2019-02-28 at 9 53 43 am

Updates!! Alignment-aware bottom corner, and bolder Button weight. It now uses `600`, with a fallback to `bold`.

Lemme know what you guys thinky

**Comment by Charca** [comment]
Thanks, man! Looks good to me 👍

**Comment by digitaldesigner** [comment]
Just to confirm that _only_ the title (i.e. the "Need Help?" link) should be Barlow, everything else San Francisco/System Font. That way we can use the same button as usual @ItsJonQ

**Comment by ItsJonQ** [comment]
@digitaldesigner Gotcha!! 👍 WIll make that update now

**Comment by ItsJonQ** [comment]
@digitaldesigner Here we go!

screen shot 2019-02-28 at 4 13 08 pm

Barlow, only on the title text :)

**Comment by ItsJonQ** [comment]
Made the final adjustments, and got approval from @digitaldesigner

**Comment by ItsJonQ** [comment]
cc'ing @ryan-mulrooney for QA review

**Comment by ryan-mulrooney** [comment]
@Charca @ItsJonQ This is looking great. Just a couple of things...

- None of the text areas have any character limiting. Is that something we should address here?
- There's no input in Storybook to QA the button fully (for changing the text).

**Comment by ItsJonQ** [comment]
@ryan-mulrooney @Charca I think character limits would be defined by the app. I don't even know what character limits we'd even have. 😅

**Comment by ItsJonQ** [comment]
screen shot 2019-03-01 at 3 33 46 pm

Adding a knob for the Greeter button example

**Comment by ItsJonQ** [comment]
🍐 'ed with @ryan-mulrooney . Got approval! Merging this up

**Pull request merged**